src/mcpdotnet/Server/McpServerOptions.cs

```diff
@@ -8,30 +8,30 @@ namespace McpDotNet.Server;
 /// protocol version.
 /// <see href="https://spec.modelcontextprotocol.io/specification/2024-11-05/basic/lifecycle/">See the protocol specification for details on capability negotiation</see>
 /// </summary>
-public record McpServerOptions
+public class McpServerOptions
 {
     /// <summary>
     /// Information about this server implementation.
     /// </summary>
-    public required Implementation ServerInfo { get; init; }
+    public required Implementation ServerInfo { get; set; }
```

Comment:
Curious what motivated this change? I defer to @PederHP but I presume he's using immutable records so that changes can be made using `with` expressions.

Reply:
Prior to this change, it was impossible to reconfigure anything the `AddMcpServer` overload taking an options configuration callback:

```csharp
AddMcpServer(this IServiceCollection services, Action<McpServerOptions>? configureOptions = null)
```

It also wasn't using the [options pattern](https://learn.microsoft.com/dotnet/core/extensions/options) which was weird because it's used elsewhere in the project. I now does, but the options pattern requires mutable option types. I think we'll need to go further and make the property types like `Implementation` mutable as well to make it easier to configure.

Reply:
> I presume he's using immutable records so that changes can be made using with expressions which is goodness in my book

with expressions are goodness, but the immutability of these types is making various things very awkward to express.